comp/forwarder/defaultforwarder/forwarder_health.go

```diff
@@ -323,6 +325,14 @@ func (fh *forwarderHealth) checkValidAPIKeys(domain string, keys []string) (apiE
 			validKey = true
 		} else {
 			fh.log.Warnf("api_key '%s' for domain %s is invalid", scrubbedAPIKey, domain)
+
+			// Trigger throttled secret refresh on invalid API key
+			result, err := fh.secrets.Refresh(false)
```

Comment:
Same here, we should add a test to cover this new use case (ie: testing that the forwarder heath does call secret.Refrech correctly).